From 816df09b3a8500916eec08e7b3c291b510df1672 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 29 Oct 2020 19:26:24 +0700
Subject: [PATCH 338/433] drm: rcar-du: Fix byte alignment check for pitch

This patch corrects the byte alignment check for pitch when
creating a frame buffer object. The alignment value in case
of using separation format (NV16, NV12 and NV21) is corrected
to the right value.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Hung Dong <hung.dong.xd@rvc.renesas.com>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_kms.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_kms.c b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
index 7ec6422..454738f 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_kms.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
@@ -165,8 +165,11 @@ rcar_du_fb_create(struct drm_device *dev, struct drm_file *file_priv,
 
 	if (rcar_du_needs(rcdu, RCAR_DU_QUIRK_ALIGN_128B))
 		align = 128;
+	else if ((format->fourcc == DRM_FORMAT_NV12) ||
+		 (format->fourcc == DRM_FORMAT_NV21))
+		align = 16;
 	else
-		align = 16 * bpp;
+		align = (16 * bpp) / format->planes;
 
 	if (mode_cmd->pitches[0] & (align - 1) ||
 	    mode_cmd->pitches[0] >= max_pitch) {
-- 
2.7.4

